File: test_data/code/python/1189-maximum-number-of-balloons.py
Python
class Solution:
    def maxNumberOfBalloons(self, text: str) -> int:
        count = {"b": 0, "a": 0, "l": 0, "o": 0}
        for char in text:
            if char in count:
                count[char] += 1
                
        min_count = float("inf")
        for char, c in count.items():
            if char == "b" or char == "a":
                min_count = min(min_count, c)
            elif char == "l":
                min_count = min(min_count, c // 2)
            elif char == "o":
                min_count = min(min_count, c // 2)
            else:
                min_count = min(min_count, c // 1)
                
        return min_count

File: test_data/code/java/1189-maximum-number-of-balloons.java
class Solution {
    public int maxNumberOfBalloons(String text) {
        int[] counts = new int[5];
        for (char c : text.toLowerCase().toCharArray()) {
            if (c == 'b') counts[0]++;
            else if (c == 'a') counts[1]++;
            else if (c == 'l') counts[2]++;
            else if (c == 'o') counts[3]++;
            else if (c == 'n') counts[4]++;
        }
        // "balloon" has 2 L's, 2 O's
        counts[2] /= 2;
        counts[3] /= 2;
        // find the min count of each character
        int minCount = Integer.MAX_VALUE;
        for (int count : counts) {
            minCount = Math.min(minCount, count);
        }
        return minCount;
    }
}